### ISSUE TITLE
Remove transaction id from context.

### ISSUE THREAD

**Issue by doczir** [emartech/scala-logger#5]
Transaction Id should not be handled as a special case, it should be added to the logging context as any other value.

* [ ] remove transaction id from logging context
* [ ] add optional method to create logging context with a transaction id `LoggingContext#withTransactionId`
* [ ] modify default apply method to create an empty logging context

**Comment by miklos-martin** [comment]
I'm not sure, it does enforce a best practice. Also, what an empty context is useful for?

I would be in favor of requiring a transaction id for constructing a context but handling it like any other value in the context.

**Comment by doczir** [comment]
An empty context is useful, if you do not have any context to log (which is kind of a bad practice, I agree). The question is if we want to support it this scenario.

Maybe we should scratch the default apply method, only allow creation with transaction id, and, as we sometimes extract it from context, we could add a utility method which retrieves transaction id from the internal map.

The main point of this issues is that having it handled differently makes the implementation of #6 a bit messy.

So in conclusion, this could be split into two separate issues:
* an internal refactor
* whether we want to support empty contexts

**Comment by tg44** [comment]
What if I specifically want to use other name as transactionId?
If we want to stay with the "no empty context" requirement I think our applys should be sth like these:
 - apply() -> Context("transactionId" -> uuid)
 - apply(name: String) -> Context(name -> uuid)
 - apply(firstFragment: (String, String)) -> Context(firstfregment)

**Comment by miklos-martin** [comment]
I don't really like the utility method idea. How about adding the transaction id to the other values instead, while encoding them for logging?
